components/service/pocket/src/main/java/mozilla/components/service/pocket/api/PocketJSONParser.kt

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.pocket.api
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.service.pocket.logger
+import mozilla.components.support.ktx.android.org.json.mapNotNull
+import mozilla.components.support.ktx.android.org.json.tryGetInt
+import mozilla.components.support.ktx.android.org.json.tryGetString
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+private const val JSON_STORY_TITLE_KEY = "title"
+private const val JSON_STORY_URL_KEY = "url"
+private const val JSON_STORY_IMAGE_URL_KEY = "imageUrl"
+private const val JSON_STORY_PUBLISHER_KEY = "publisher"
+private const val JSON_STORY_CATEGORY_KEY = "category"
+private const val JSON_STORY_TIME_TO_READ_KEY = "timeToRead"
+
+/**
+ * Holds functions that parse the JSON returned by the Pocket API and converts them to more usable Kotlin types.
+ */
+internal class PocketJSONParser internal constructor() {
```

Comment:
nit/question: If the class is already internal, what does an internal constructor add? 🤔